src/sanitize/sanitizeNumbers.ts

```diff
@@ -49,7 +59,16 @@ function sanitizeCodec(value: Codec): AnyJson {
 			if (!property) {
 				return jsonStruct;
 			}
-			jsonStruct[key] = sanitizeNumbers(property);
+
+			jsonStruct[key] = sanitizeNumbers(property, options);
```

Comment:
Is it guaranteed that, by the time you're looking at the "value" key in `sanitizeMetaDataExceptions` below, the "type" key has definitely already been sanitized and placed in the `jsonStruct`?

(if `options.isMetadata`, do we know that the shape of the struct is something like `{ type, value }`? I wonder whether it makes any sense to duplicate a little more logic so that it's something like:

```
if (value instanceof Struct) {
   if (options?.isMetadata && options?.registry) {
      return sanitizeMetadataStruct(value);
   } else {
      return value.defKeys.reduce((jsonStruct, key) => { ... });
   }
}
```

Just so that all of the important logic/information for "handle this random struct" or "handle this specific type of struct" is in one place and neatly separated? (I might be misunderstanding though and this might make no sense!)

Reply:
(I guess in any case, a commentalong the lines of "we expect the metadata struct to look like `{ key: string, value: Bytes }`, and we want to convert that value into something more meaningful like a u8/u16.." would help me here because then I'd be like "ah ok, I see, we're looking for something of that shape, gotcha!)

Reply:
Based on our conversation yesterday, resolving this. I also added some inline notes for clarifications, and another test for `Option<u128>`. This is a good test because it will break if the creating of the type is ever done directly from the hex value (we need to create each type by its Uint8Array).